src/main/java/org/betonquest/betonquest/objectives/EnchantObjective.java

```diff
@@ -67,52 +84,39 @@ public void stop() {
         HandlerList.unregisterAll(this);
     }
 
-    @Override
-    public String getDefaultDataInstruction() {
-        return "";
-    }
-
-    @Override
-    public String getProperty(final String name, final Profile profile) {
-        return "";
-    }
-
-    public static class EnchantmentData {
-
-        private final Enchantment enchantment;
-        private final int level;
-
-        public EnchantmentData(final Enchantment enchantment, final int level) {
-            this.enchantment = enchantment;
-            this.level = level;
-        }
-
+    /**
+     * Represents an enchantment and its level.
+     *
+     * @param enchantment the enchantment
+     * @param level       the level
+     */
+    public record EnchantmentData(Enchantment enchantment, int level) {
+
+        /**
+         * Converts user input to an EnchantmentData object.
+         *
+         * @param string the string to parse
+         * @return the parsed EnchantmentData object
+         * @throws InstructionParseException if the user defined string is not a valid enchantment or does not
+         *                                   contain a level
+         */
         @SuppressWarnings({"deprecation", "PMD.AvoidLiteralsInIfCondition"})
         public static EnchantmentData convert(final String string) throws InstructionParseException {
             final String[] parts = string.split(":");
-            if (parts.length != 2) {
-                throw new InstructionParseException("Could not parse enchantment: " + string);
-            }
+            //TODO: v3 / Config Patcher: Use Enchantment.getByKey for real Minecraft enchantment names
```

Comment:
This TODO may be controversal.

Reply:
do not use the todo here. If you want to document it, make in issue or add it to the related entry in the objectives discussion